src/View/mainpage/DrawROIWindow/Drawing.py

```diff
@@ -67,6 +73,10 @@ def __init__(self, imagetoPaint, pixmapdata, min_pixel, max_pixel, dataset,
         self.max_bounds_y = self.cols
         self.fill_source = None
         self.is_drawing = False
+        if 'xy' in kwargs:
+            self.fill_source = kwargs.get('xy')
+        if 'UI' in kwargs:
+            self.seed.connect(kwargs.get('UI').set_seed)
```

Comment:
consider creating the signal above and connecting, then passing the signal down so that this object can emit.
Or something similar, try to avoid passing down the parent obj.

Reply:
Code was written by Jack and I can't fix that as I don't know enough about how it works